### PR TITLE
Add docs for the different trees

### PR DESCRIPTION
**Pull request by eparejatobes**
Addresses #6.

### CODE REVIEW

File: docs/readme.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Taxonomy
+
+## Unclassified Tree
+
+Nodes are deemed *unclassified* when their *scientific name* contains (ignore case) the word `unclassified`; these generate the **Unclassified** tree.
```
Comment (eparejatobes) [inline docs/readme.md]: @agarciamontoro I'm assuming the meaning of *generate* here is common knowledge; add it here if you want.

File: docs/readme.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+## Environmental Samples Tree
+
+The **Environmental Samples** tree is generated by the nodes whose *scientific name* has prefix (ignore case) `environmental samples`.
```
Comment (eparejatobes) [inline docs/readme.md]: Note that *scientific name* is the one and only name that api.ncbitaxonomy gives you.

**Comment by eparejatobes** [comment]
Addresses #4 too.

File: docs/readme.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Taxonomy
+
```
Comment (agarciamontoro) [inline docs/readme.md]: We may add a minimal intro here describing that we are generating two more trees out of the original one, picking nodes that meet some criteria (unclassified in one case, environmental samples in the other) and "extracting" all its ancestors and all its descendants.
I will add this after the other comments are resolved.

File: docs/readme.md (hunk 1 above)
Comment (agarciamontoro) [reply docs/readme.md]: It's ok as it is now. Previous comment addresses this.

File: docs/readme.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+Nodes are deemed *unclassified* when their *scientific name* contains (ignore case) the word `unclassified`; these generate the **Unclassified** tree.
+
+## Environmental Samples Tree
+
+The **Environmental Samples** tree is generated by the nodes whose *scientific name* has prefix (ignore case) `environmental samples`.
+
+# File structure
+
+Each relation is stored as two `csv` files (in and out): first column source, second column target. We have the following relations
```
Comment (agarciamontoro) [inline docs/readme.md]: I don't understand the *(in and out)*. Could you add an example of one of the trees specifying its set of files?

File: docs/readme.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+1. `parent` (the tree structure)
+2. `scientific_name`
+3. `rank`
+
+For all these entities we use their obvious `String` representation.
+
+Of course, we have one set of this files for each of the trees we provide (full, good, unclassified, environmental samples).
```
Comment (agarciamontoro) [inline docs/readme.md]: How do we define the *good tree*? The full tree minus the (*unclassified* or *environmental samples*) nodes branches? Should we keep the ancestors of the *bad* nodes in the *good tree*?

(*The good tree* sounds like something biblical :angel:)

File: docs/readme.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+## Tree mappings
+
+The inclusions of the other trees in the full one could be represented in the same way as with any other relation, the source and target being the same ID. Alternatively, we could represent the corresponding membership predicate.
```
Comment (agarciamontoro) [inline docs/readme.md]: I don't understand the *source and target being the same ID* either, but I guess I will when I understand what you meant with the relations and the csv files above.

File: docs/readme.md (hunk 4 above)
Comment (agarciamontoro) [reply docs/readme.md]: Ok, now it seems obvious :man_facepalming:, my bad.

And the last question: for `scientific_name` and `rank`, do we also have two CSVs? One file with the node ID in the first conlumn and their `scientific_name` (resp. `rank`) in the second one, and the other file with the `scientific_name` (resp. `rank`) in the first column and all (?) node IDs with that `scientific_name` (resp. `rank`) in the second one?

Comment (eparejatobes) [reply docs/readme.md]: Yes; index as in "in" for "properties".

**Comment by agarciamontoro** [comment]
Please, check my additions and merge if everything looks ok :)

**Pull request merged**